finetuner/experiment.py

```diff
@@ -117,6 +120,79 @@ def delete_runs(self):
         """Delete every run inside the experiment."""
         self._client.delete_runs(experiment_name=self._name)
 
+    def _prepare_data(
+        self,
+        train_data: Union[DocumentArray, str],
+        run_name: str,
+        eval_data: Optional[Union[DocumentArray, str]] = None,
+        eval_callback: Optional[EvaluationCallback] = None,
+    ):
+        """Upload data to Hubble and returns their names.
+
+        Uploads all data needed for fine-tuning - training data,
+        evaluation data and query/index data for `EvaluationCallback`.
+
+        Checks not to upload same dataset twice.
+
+        :param train_data: Either a `DocumentArray` for training data or a
+            name of the `DocumentArray` that is pushed on Hubble.
+        :param run_name: Name of the run.
+        :param train_data: Either a `DocumentArray` for evaluation data or a
+            name of the `DocumentArray` that is pushed on Hubble.
+        :param eval_callback: Evaluation callback that contains query and index data
+            and gets modified in-place.
+        :return: Name(s) of the uploaded data.
+        """
+        train_data_name = push_data_to_hubble(
```

Comment:
We make too many assumptions in this code block. Query data should be required. If not provided raise a value error. The only assumption we should make is that if index data are not given then query data == index data. And you don't even need to implement this. It is happening in the run